spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/tf/loaders/Adapter.scala

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.intel.analytics.bigdl.utils.tf.loaders
+
+import com.intel.analytics.bigdl.nn.abstractnn.{AbstractModule, Activity}
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.utils.{T, Table}
+
+import scala.reflect.ClassTag
+
+/**
+ * Sometimes parameters pass as tensor from some previous nodes in graph. So we can't construct
+ * module at module define time. In such case, you can use the AdapterOperation to wrapper the
+ * module and create it in runtime.
+ *
+ * Please note you must guarantee the input parameter won't change each time.
```

Comment:
Should we check this in our code? As it will be very hard to debug if the input parameter do change.

Reply:
fixed